jujugui/static/gui/src/app/utils/bakery-utils.js

```diff
@@ -85,40 +86,65 @@ YUI.add('bakery-utils', function(Y) {
         plans: config.plansURL,
         terms: config.termsURL
       }
-    });
-    return new jujulib.Bakery(webHandler, storage, {
-      nonInteractive: !config.interactiveLogin,
-      visitPage: url => {
-        // Add to the page a notification about accepting the pop up window
-        // for logging into USSO.
-        const holder = document.getElementById('login-notification');
-        const content = (
+    }); const interactiveVisit = error => {
+      // Add to the page a notification about accepting the pop up window
+      // for logging into USSO.
+      const url = error.Info.VisitURL;
+      const holder = document.getElementById('login-notification');
+      const content = (
           <span>
-            To proceed with the authentication, please accept the pop up
-            window or&nbsp;
-            <a href={url} target="_blank">click here</a>.
+          To proceed with the authentication, please accept the pop up
+        window or&nbsp;
+          <a href={url} target="_blank">click here</a>.
           </span>);
-        let dismiss = null;
-        if (stateGetter().root !== 'login') {
-          dismiss = () => {
-            ReactDOM.unmountComponentAtNode(holder);
-          };
-        }
-        ReactDOM.render(
+      let dismiss = null;
+      if (stateGetter().root !== 'login') {
+        dismiss = () => {
+          ReactDOM.unmountComponentAtNode(holder);
+        };
+      }
+      ReactDOM.render(
           <window.juju.components.Notification
-            dismiss={dismiss}
-            content={content}
+        dismiss={dismiss}
+        content={content}
           />, holder);
-        // Open the pop up (default behavior for the time being).
-        window.open(url, 'Login');
-      },
+      // Open the pop up (default behavior for the time being).
+      window.open(url, 'Login');
+    };
+
+    const nonInteractiveVisit = error => {
+      const JSON_CONTENT_TYPE = 'application/json';
+      const acceptHeaders = {'Accept': JSON_CONTENT_TYPE};
+      const contentHeaders = {'Content-Type': JSON_CONTENT_TYPE};
+      const login = function(response) {
+        var method = error.jujugui;
+        var data = JSON.stringify({login: config.auth});
+        return webHandler.sendPostRequest(
+          method, contentHeaders, data,
+          null, null, false, null, null);
+      };
+
+      return webHandler.sendGetRequest(
+        error.Info.VisitURL,
+        acceptHeaders, null, null, false, null, login.bind(this));
+    };
+
+    let params = {
       onSuccess: () => {
```

Comment:
Technically this specific onSuccess should be only provided if `config.interactiveLogin` is true. So maybe it could be part of the conditional block below?